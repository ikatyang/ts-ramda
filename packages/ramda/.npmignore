node_modules/
scripts/
src/
templates/
.npmignore
test.ts
tsconfig.json
tslint.json
