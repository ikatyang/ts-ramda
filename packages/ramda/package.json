{
  "name": "@ikatyang/types-ramda",
  "version": "0.23.0",
  "description": "TypeScript's type definitions for Ramda",
  "repository": "https://github.com/ikatyang/types",
  "author": "ikatyang",
  "license": "MIT",
  "scripts": {
    "ts-node": "../../node_modules/.bin/ts-node -P ./tsconfig.json",
    "build": "yarn run ts-node -- ./scripts/build.ts ./templates ./typings"
  },
  "devDependencies": {
    "ramda": "0.23.0"
  }
}
