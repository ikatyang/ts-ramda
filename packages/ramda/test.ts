/// <reference path="./index.d.ts" />

import * as R from 'ramda';

R.__;

R.add(1, 2);
R.add(1)(2);
