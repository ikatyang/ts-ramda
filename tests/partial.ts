import * as R from 'ramda';

let multiply = function(a: number, b: number) { return a * b; };
let double = R.partial(multiply, [2]);
// @dts-jest
double(2); // => 4

let greet = function(salutation: string, title: string, firstName: string, lastName: string) {
  return salutation + ', ' + title + ' ' + firstName + ' ' + lastName + '!';
};
let sayHello = R.partial(greet, ['Hello']);
let sayHelloToMs = R.partial(sayHello, ['Ms.']);
// @dts-jest
sayHelloToMs('Jane', 'Jones'); // => 'Hello, Ms. Jane Jones!'

let greetMsJaneJones = R.partialRight(greet, ['Ms.', 'Jane', 'Jones']);
// @dts-jest
greetMsJaneJones('Hello'); // => 'Hello, Ms. Jane Jones!'
