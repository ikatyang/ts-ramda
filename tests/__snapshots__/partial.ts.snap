// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`double(2) 1`] = `"number"`;

exports[`greetMsJaneJones('Hello') 1`] = `"string"`;

exports[`sayHelloToMs('Jane', 'Jones') 1`] = `"string"`;
